rfc-012-makefile-standards.md

```diff
@@ -0,0 +1,73 @@
+# Makefile standards
+
+## Summary
+
+A set of standards for Makefiles across all projects
+
+## Problem
+
+When setting up/jumping onto a new project - sometimes it can be hard to discover what commands
+need to be run to perform common tasks (e.g. Running tests, serving the local development environment).
+This can make switching between projects more difficult as there can be a large number of ways to perform
+all of these tasks.
+
+Having a standard set of commands that can be run across all projects for common tasks
+allows people to get projects up and running without having to spend time looking into
+how things are set up to know what specific commands to run.
+
+
+## Proposal
+
+All projects **MUST** have a Makefile.
+
+All projects **MUST** have relevant [`PHONY` targets](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html)
+
+**All** projects **MUST** have the commands:
+- `make test`
+  - Runs the tests for the project
+  - This **SHOULD** run the tests in a 'watch' mode (e.g. [Guard](https://github.com/guard/guard))
```

Comment:
I disagree with this, personally. Would prefer `test-watch` or similar for watching tests.

Reply:
Yeah I was 50/50 on mentioning test watching as a requirement, hence the should. Maybe move it to `MAY` as opposed to `SHOULD`?

Reply:
Not sure where all my previous comments have gone 😨 but potentially add an additional task `test-watch` or something similar if that's what you actually want.

Reply:
The way Jest works is it looks for a `CI=true` environment variable

Reply:
How can we progress with a decision on this point? My vote would be for it to not default to watching, but for there to be an additional task for this.

Reply:
Vote for MAY watch, or additional command.